### PR TITLE
IOTPROT-860 Duplicate detector interface

### PR DESCRIPTION
**Pull request by samhaa01**

### CODE REVIEW

File: coap-core/src/main/java/com/mbed/coap/server/internal/DuplicationDetectorImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package com.mbed.coap.server.internal;
+
+import com.mbed.coap.packet.CoapPacket;
+import com.mbed.coap.server.CoapRequestId;
+import com.mbed.coap.server.PutOnlyMap;
+
+/**
+ * Checks if incoming request has been repeated
+ *
+ * @author szymon
```
Comment (tuuket01) [inline coap-core/src/main/java/com/mbed/coap/server/internal/DuplicationDetectorImpl.java:25]: I don't think the author is szymon :wink:

File: coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java [hunk 2]
```diff
@@ -310,6 +314,11 @@ public CoapServerBuilderForUdp duplicateMessageDetectorCache(PutOnlyMap<CoapRequ
             return this;
         }
 
+        public CoapServerBuilderForUdp duplicateMessageDetector(DuplicationDetector detector) {
```
Comment (norbert-david) [inline coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java:317]: Internal interface is exposed on the public builder.

**Review by norbert-david — CHANGES_REQUESTED** [review]
So we have now DuplicateDetectorCache and DuplicateDetector as well, separately?

File: coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java (hunk 2 above)
Comment (samhaa01) [reply coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java:316]: I think DuplicateDetector could be public interface. Moved it to public.

File: coap-core/src/main/java/com/mbed/coap/server/internal/DuplicationDetectorImpl.java (hunk 1 above)
Comment (samhaa01) [reply coap-core/src/main/java/com/mbed/coap/server/internal/DuplicationDetectorImpl.java]: fixed.

**Review by norbert-david — APPROVED** [review]

**Review by tuutep01 — CHANGES_REQUESTED** [review]
This is not needed.

**Pull request closed without merging**